src/basic/socket-util.c

```diff
@@ -399,19 +399,25 @@ int sockaddr_pretty(
                         if (r < 0)
                                 return -ENOMEM;
                 } else {
-                        char a[INET6_ADDRSTRLEN];
+                        char a[INET6_ADDRSTRLEN], ifname[IF_NAMESIZE + 1];
 
                         inet_ntop(AF_INET6, &sa->in6.sin6_addr, a, sizeof(a));
+                        if (sa->in6.sin6_scope_id != 0)
```

Comment:
i wonder if we should conditionalize this further: check if this is from the ipv6 link-local range, and if not suppress it to, since it's ignored by the kernel then iirc

Reply:
it's even documented in ipv6(7)

Reply:
I'd say that the printer should not do the suppression. It should faithfully report what is in the struct. (It's just confusing if the printer is too smart.) IMHO, instead we could detect such misuse in the parser, and warn and ignore. This would also be much more useful for users.